x/gamm/keeper/pool_service_test.go

```diff
@@ -222,6 +242,181 @@ func (suite *KeeperTestSuite) TestCreateBalancerPool() {
 	}
 }
 
+func (suite *KeeperTestSuite) TestInitializePool() {
+	params := suite.App.GAMMKeeper.GetParams(suite.Ctx)
+	testAccount := suite.TestAccs[0]
+
+	// get raw pool creation fee(s) as DecCoins
+	poolCreationFeeDecCoins := sdk.DecCoins{}
+	for _, coin := range params.PoolCreationFee {
+		poolCreationFeeDecCoins = poolCreationFeeDecCoins.Add(sdk.NewDecCoin(coin.Denom, coin.Amount))
+	}
+
+	tests := []struct {
+		name        string
+		msg         swaproutertypes.CreatePoolMsg
+		usingMock   bool
+		emptySender bool
+		expectPass  bool
+	}{
+		{
+			name:        "initialize balancer pool with default assets",
+			msg:         balancer.NewMsgCreateBalancerPool(testAccount, defaultPoolParams, defaultPoolAssets, defaultFutureGovernor),
+			emptySender: false,
+			expectPass:  true,
+		},
+		{
+			name:        "initialize stableswap pool with default assets",
+			msg:         stableswap.NewMsgCreateStableswapPool(testAccount, defaultStableSwapPoolParams, defaultStableSwapPoolAssets, defaultScalingFactor, ""),
+			emptySender: false,
+			expectPass:  true,
+		},
+		{
+			name:      "initialize a CL pool which cause panic",
+			msg:       balancer.NewMsgCreateBalancerPool(testAccount, defaultPoolParams, defaultPoolAssets, defaultFutureGovernor),
```

Comment:
using balancer `MsgCreatePool` is misleading. Can you try this instead: https://github.com/osmosis-labs/osmosis/blob/f611e01c7fb9ab9d5348250cb0b65e08f3ec651b/x/concentrated-liquidity/model/msgs.go#L24

If you use it, you might find it easier to move away from using mocks

Reply:
If I use ``NewMsgCreateConcentratedPool``, I will still have to do a little trick because ``swaprouterKeeper.GetPoolModule`` only return if the pool id is registered which only happens in constructor process

Reply:
Or instead of using ``NewMsgCreateConcentratedPool``, I can use a mock.

Reply:
Instead of ``NewMsgCreateConcentratedPool``, I can use a mock. I think it is more appropriate to use a mock. Can you check it again @p0mvn ?